receiver/receivercreator/receiver.go

```diff
@@ -55,94 +57,41 @@ func new(logger *zap.Logger, nextConsumer consumer.MetricsConsumerOld, cfg *Conf
 	return r, nil
 }
 
-// loadRuntimeReceiverConfig loads the given subreceiverConfig merged with config values
-// that may have been discovered at runtime.
-func (rc *receiverCreator) loadRuntimeReceiverConfig(factory component.ReceiverFactoryOld,
-	staticSubConfig *subreceiverConfig, discoveredConfig userConfigMap) (configmodels.Receiver, error) {
-	mergedConfig := config.NewViper()
-
-	// Merge in the config values specified in the config file.
-	if err := mergedConfig.MergeConfigMap(staticSubConfig.config); err != nil {
-		return nil, fmt.Errorf("failed to merge subreceiver config from config file: %v", err)
-	}
-
-	// Merge in discoveredConfig containing values discovered at runtime.
-	if err := mergedConfig.MergeConfigMap(discoveredConfig); err != nil {
-		return nil, fmt.Errorf("failed to merge subreceiver config from discovered runtime values: %v", err)
-	}
-
-	viperConfig := config.NewViper()
-
-	// Load config under <receiver>/<id> since loadReceiver and CustomUnmarshaler expects this structure.
-	viperConfig.Set(staticSubConfig.fullName, mergedConfig.AllSettings())
-
-	receiverConfig, err := config.LoadReceiver(mergedConfig, staticSubConfig.receiverType, staticSubConfig.fullName, factory)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load subreceiver config: %v", err)
-	}
-	// Sets dynamically created receiver to something like receiver_creator/1/redis{endpoint="localhost:6380"}.
-	// TODO: Need to make sure this is unique (just endpoint is probably not totally sufficient).
-	receiverConfig.SetName(fmt.Sprintf("%s/%s{endpoint=%q}", rc.cfg.Name(), staticSubConfig.fullName, mergedConfig.GetString("endpoint")))
-	return receiverConfig, nil
+// safeHost provides a safer version of host for receivers started at runtime.
```

Comment:
Let's calling this something like `loggingHost`. "safety" can mean different things depending on the requirement. In this case the requirement is that fatal errors are logged instead of crashing the process. But crashing the process is the "safe" option in some other case (during startup).